model/implementation/DeliveryExecutionStateService.php

```diff
@@ -20,11 +20,15 @@
 
 namespace oat\taoProctoring\model\implementation;
 
+use DateInterval;
+use DateTime;
+use DateTimeImmutable;
 use oat\oatbox\service\ConfigurableService;
-use oat\taoQtiTest\models\TestSessionMetaData;
+use oat\taoAct\model\testSession\TestSessionProcessor;
```

Comment:
My opinion is that it is not necessary to add new dependency and use `TestSessionProcessor` here. You should not pass exit codes to the `DeliveryExecutionTerminated` event. You can implement new custom event (something like `DeliveryExecutionExpired`) and add listener of this event to the `TestSessionProcessor`. The reason of creation `TestSessionProcessor` was to let him decide for themselves which exit code should be saved, and encapsulate all work with exit codes in it.